README.md

```diff
@@ -44,6 +44,22 @@ uv run python3 src/scraper.py
 
 We welcome contributions to enhance Courses API! If you find any issues, have suggestions, or want to request a feature, please follow our [Contributing Guidelines](https://github.com/compsci-adl/.github/blob/main/CONTRIBUTING.md).
 
+This repository is configured with a pre-commit linting and formatting tool
+To perform a commit from local, pre-commit must first be installed:
+```
+pip3 install pre-commit
+```
+OR if performing the action in bash:
+```
+python3 -m pip install pre-commit
+```
```

Comment:
We want to have pre-commit installed at a project level, so you can just do `uv add pre-commit --dev` and pre-commit should be installed when you do `uv sync`. You still need to do `pre-commit install` though to install the hook.